src/journal/Journaler.h

```diff
@@ -44,6 +45,7 @@ class Journaler {
   typedef cls::journal::Tag Tag;
   typedef std::list<cls::journal::Tag> Tags;
   typedef std::set<cls::journal::Client> RegisteredClients;
+  typedef JournalMetadata::Listener JournalMetadataListener;
```

Comment:
Minor: maybe move this out to its own file now that it is being exposed outside of libjournal? Then you can include the listener header in ```librbd/Journal.h``` and define the concrete class in the header.